README.md

```diff
@@ -2,31 +2,739 @@
+
+
+## Aggregate Functions  (Code Along)
+
+We will work with the same `pet` table from the previous lesson.  
+
+SQL provides functions for aggregating values across multiple rows:
+
+- `COUNT(col)` - return a count of non-null values in column `col`.
+- `SUM(col)` - return the sum of values in column `col`.
+- `MIN(col)` - return the minimum value in column `col`.
+- `MAX(col)` - return the maximum value in column `col`.
+- `AVG(col)` - return the average value in column `col`.
+
+
+<table>
+<tr>
+<th>
+Query
+</th>
+<th>
+Result
+</th>
+</tr>
+
+<tr>
+<td>
+Count all rows:
+
+```sql
+SELECT COUNT(*)
+FROM pet;
+```
+
+</td>
+<td>
+
+```text
+count
+
+8
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+Count rows
+with a non-null value in name:
+
+```sql
+SELECT COUNT(name)
+FROM pet;
+```
+
+</td>
+<td>
+
+```text
+count
+
+8
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+Count rows
+with a non-null value in age:
+
+```sql
+SELECT COUNT(age)
+FROM pet;
+```
+
+</td>
+<td>
+
+```text
+count
+
+7
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+Count of distinct species:
+
+```sql
+SELECT COUNT(DISTINCT species)
+FROM pet;
+```
+
+</td>
+<td>
+
+```text
+count
+
+3
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+Count the number of dogs:
+
+```sql
+SELECT COUNT(*)
+FROM pet
+WHERE species = 'dog';
+```
+
+</td>
+<td>
+
+```text
+count
+
+4
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+Sum the ages of all pets:
+
+```sql
+SELECT SUM(age)
+FROM pet;
+```
+
+</td>
+<td>
+
+```text
+sum
+
+35
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+Sum the ages of Tabby cats:
+
+```sql
+SELECT SUM(age)
+FROM pet
+WHERE breed = 'Tabby' and species = 'cat' ;
+```
+
+</td>
+<td>
+
+```text
+sum
+
+11
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+Minimum, maximum and average age
+among all pets:
+
+```sql
+SELECT MIN(age), MAX(age), AVG(age)
+FROM pet;
+```
+
+</td>
+<td>
+
+```text
+min max avg
+
+1   12  5    
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+Minimum, maximum and average age
+among all cats:
+
+```sql
+SELECT MIN(age), MAX(age), AVG(age)
+FROM pet
+WHERE species = 'cat';
+```
+
+</td>
+<td>
+
+```text
+min max avg
+
+1   10  5.333333333
+```
+
+</td>
+</tr>
+
+
+<tr>
+<td>
+Truncate to display 2 digits after decimal point:
+
+```sql
+SELECT MIN(age), MAX(age), TRUNC(AVG(age),2) 
+FROM pet
+WHERE species = 'cat';
+```
+
+</td>
+<td>
+
+```text
+min max avg
+
+1   10  5.33
+```
+
+</td>
+</tr>
+
+</table>
+
+
+
+## GROUP BY
+
+The `GROUP BY` clause collects rows with the same value for a column into a group.
+
+
+```text
+id  name        species	breed	            age
+
+1   Moe         cat     Tabby	            10
+2   Hana        cat     Tabby	            1
+3   Lil' Bub    cat     American Shorthair  5
+4   Snuggles    dog     Bichon	            5
+5   Honey       dog     Cavachon            NULL
+6   Herbie      fish    NULL                4
+7   Fifi        dog     Poodle              1
+8   Lulu        dog     Yorkiepoo           12
+```
```

Comment:
Okay, my SELECT * FROM pet statement shows this same exact table. So that is good!

Reply:
good!